pyproject.toml

```diff
@@ -12,22 +12,26 @@ classifiers = [
     'Programming Language :: Python :: 3.9',
     'Programming Language :: Python :: 3.10',
     'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
     'Topic :: Scientific/Engineering :: Artificial Intelligence',
 ]
 keywords = [ 'copulas' ]
 dynamic = ["version"]
 license = { text = 'BSL-1.1' }
-requires-python = '>=3.8,<3.12'
+requires-python = '>=3.8,<3.13'
 readme = 'README.md'
 dependencies = [
     "numpy>=1.20.0,<2;python_version<'3.10'",
-    "numpy>=1.23.3,<2;python_version>='3.10'",
+    "numpy>=1.23.3,<2;python_version>='3.10' and python_version<'3.12'",
+    "numpy>=1.26.0,<2;python_version>='3.12'",
     "pandas>=1.1.3;python_version<'3.10'",
     "pandas>=1.3.4;python_version>='3.10' and python_version<'3.11'",
     "pandas>=1.5.0;python_version>='3.11'",
     'plotly>=5.10.0,<6',
     "scipy>=1.5.4,<2;python_version<'3.10'",
-    "scipy>=1.9.2,<2;python_version>='3.10'",
+    "scipy>=1.9.2,<2;python_version>='3.10' and python_version<'3.12'",
+    "scipy>=1.12.0,<2;python_version>='3.12'",
+    'setuptools<70',
```

Comment:
If we are adding setuptools because the __init__ file uses pkg_resources, I think we should just switch to importlib instead since pkg_resources is deprecated. @frances-h I remember we wanted to do this before but had to keep pkg_resources for some reason. Was it because of python 3.8 or 3.7?

Reply:
If I remember correctly it was Python 3.7 because `importlib.metadata` was added in Python 3.8.